lib/index.d.ts

```diff
@@ -0,0 +1,600 @@
+            sellerInputIdentifier: string,
+            isAmazonFulfilled: boolean,
+            idValue: string,
+            priceToEstimateFees: {
+                listingPrice: {
+                    amount: string,
+                    currencyCode: string,
+                },
+                shipping: {
+                    amount: string,
+                    currencyCode: string,
+                },
+            },
+        },
+        status: 'ClientError' | 'ServerError' | string, // TODO: not sure what a value here is with no error
+        // totalFees only if no error
+        totalFees?: {
+            amount: string,
+            currencyCode: string,
+        },
+        // time only if no error
+        time?: string,
+        // detail only if no error, need to fill this in
+        detail?: Array<any>,
+        error?: {
+            code: string,
+            message: string,
+            type: 'Sender' | 'Receiver',
+        },
+    },
+};
+
+type NextTokenParams = { NextToken: string };
+
+type RequestReportParams = {
+    ReportType: string,
+    ReportProcessingStatus?: string,
+    EndDate?: string,
+    Scheduled?: boolean,
+    ReportRequestId?: string,
+    SubmittedDate?: string,
+    StartDate?: string,
+};
+type RequestReportReturn = any;
+
+type GetReportRequestListParams = {
+    ReportRequestIdList?: Array<string>,
+    ReportTypeList?: string,
+    ReportProcessingStatusList?: string,
+    MaxCount?: number,
+    RequestedFromDate?: Date,
+    RequestedToDate?: Date,
+};
+type GetReportRequestListReturn = {
+    nextToken?: string,
+    reportRequestList: Array<{
+        ReportType: string,
+        ReportProcessingStatus: string,
+        EndDate: string,
+        Scheduled: boolean,
+        ReportRequestId: string,
+        StartedProcessingDate: string,
+        StartDate: string,
+        CompletedDate: string,
+        GeneratedReportId: string,
+    }>,
+};
+
+type GetReportParams = { ReportId: string };
+type GetReportReturn = any;
+
+type GetReportListParams = {
+    MaxCount?: number,
+    ReportTypeList?: Array<string>,
+    Acknowledged?: boolean,
+    ReportRequestIdList?: Array<string>,
+    AvailableFromDate?: Date,
+    AvailableToDate?: Date,
+};
+type GetReportListReturn = any;
+
+type RequestAndDownloadReportReturn = any;
+
+type ManageReportScheduleParams = {
+    ReportType: string,
+    Schedule: any,
+    ScheduleDate?: Date,
+};
+type ManageReportScheduleReturn = any;
+
+type UpdateReportAcknowledgementsParams = {
+    ReportIdList: Array<string>,
+    Acknowledged?: boolean,
+};
+type UpdateReportAcknowledgementsReturn = any;
+
+type GetReportScheduleListParams = { ReportTypeList?: Array<string> };
+type GetReportScheduleListReturn = any;
+
+type IdTypes =
+    | 'ASIN'
+    | 'GCID'
+    | 'SellerSKU'
+    | 'UPC'
+    | 'EAN'
+    | 'ISBN'
+    | 'JAN';
+
+type AdvIdTypes =
+    | 'asin'
+    | 'gcid'
+    | 'sellersku'
+    | 'upc'
+    | 'ean'
+    | 'isbn'
+    | 'jan';
```

Comment:
This seems like an Amazon response inconsistency. I think it would be best that this package does not expose the user to these inconsistencies. Perhaps we can return an ALL CAPS version to the user?

Reply:
hmm.  Looks like GetFees takes the capitalized version as both an in and an out, GetMatchingProductForId takes it as an input .. and GetMatchingProductForId outputs in lower case.  I'm not sure if that's coming straight from the API or if we're parsing it into lower case.  As it's been literally a year or more since I've done any work with much of this, I'm having a little trouble following what I did here to trace it back :-S 

I think I'll file an issue on this and look into it separately, I want to get this into the master branch at least, before I make any changes, and then do a big set of changes to remove the static interface, promote some consistency, etc.

Reply:
https://github.com/ericblade/mws-advanced/issues/175